litestar/channels/backends/psycopg.py

```diff
@@ -25,6 +25,7 @@ async def on_startup(self) -> None:
         await self._exit_stack.enter_async_context(self._listener_conn)
 
     async def on_shutdown(self) -> None:
+        await self._listener_conn.close()
```

Comment:
Hm. I think this shouldn't be needed. We add `self._listener_conn` to `self._exit_stack.enter_context` in `on_startup` and close the exit stack in `on_shutdown`. That should already close the connection.

Reply:
yes you're right, my bad, loked at it too fast
just fr my understanding, any readon why you'r using a stack ?
seems like it works alright with just :
```
class PsycoPgChannelsBackend(ChannelsBackend):
    _listener_conn: psycopg.AsyncConnection

    def __init__(self, pg_dsn: str) -> None:
        self._pg_dsn = pg_dsn
        self._subscribed_channels: set[str] = set()
        # self._exit_stack = AsyncExitStack()

    async def on_startup(self) -> None:
        self._listener_conn = await psycopg.AsyncConnection.connect(self._pg_dsn, autocommit=True)
        # await self._exit_stack.enter_async_context(self._listener_conn)

    async def on_shutdown(self) -> None:
        await self._listener_conn.close()
        # await self._exit_stack.aclose()
```

Reply:
I think when I started out I had more than one contextmanager to close. Other than that, no particular reason :shrug: